lib/docs.js

```diff
@@ -256,6 +257,32 @@
  * @property {object} text.width - The width of the text, including truncation and ellipsis.
  * @property {object} text.value - The value of the text, including truncation and ellipsis.
  */
+/**
+ * The additional function to invoke when a cell is filled. The return value of this
+ * function sets the new data of the cell.
+ * @function
+ * @name canvasDataGrid.fillCellCallback
+ * @param {object} args - Arguments that are passed to the function.
+ * @param {array} args.rows - Data that is being filled into all the cells, including this one.
+ * @param {string} args.direction - Direction of the fill operation, which can be 'horizontal' for x axis, 'vertical' for y axis, or 'both' when it goes both ways, e.g., when pasting unevenly.
+ * @param {object} args.newRowData - New row data that is going to replace 'existingRowData'. The result of this function will be kept in this object.
+ * @param {object} args.existingRowData - Existing data of the row. This will be replaced with 'newRowData'.
+ * @param {number} args.rowIndex - Index of the row on the grid.
+ * @param {number} args.rowPosition - Current position/index of the 'rows'. Starts with '0'.
+ * @param {array} args.cells - All the cells, including this one, in the row. This is the same as accessing 'rows' using 'rowPosition'.
+ * @param {boolean} args.reversed - Whether the fill operation is applied in the reverse order, meaning if horizontal, we are going from right to left, and if vertical, we are going from bottom to top.
+ * @param {boolean} args.fillingRow - Whether the current row is being filled. If the 'direction' is 'vertical', this will always be true, or sometimes true if the 'direction' is 'both'.
```

Comment:
This is something I've been mulling for a bit, and it's yet to appear in a contributing style guide anywhere, but I would like to propose that for newly introduced `true/false` variables, where it makes sense and is possible, we name them along the lines of `hasFoo` or `isBlah` or `<verb>Something`, such that `reversed` becomes `isReversed` and `fillingRow` becomes something like `isFillingRow` or `fillsRow` — as a clearer indication that it's a boolean flag. Now for instance, reading `fillingRow` it seems to that's it's the row itself, not a boolean. Turning it into `isFillingRow` or `fillsRow` makes it more obvious, I feel.

What do you think?